src/GitVersion.Core.Tests/Formatting/LegacyFormatterProblemTests.cs

```diff
@@ -0,0 +1,163 @@
+using GitVersion.Core.Tests.Helpers;
+using GitVersion.Formatting;
+
+namespace GitVersion.Core.Tests.Formatting;
+
+[TestFixture]
+public class LegacyFormatterProblemTests
+{
+    private TestEnvironment environment;
+
+    [SetUp]
+    public void Setup() => environment = new TestEnvironment();
+
+    // ==========================================
+    // PROBLEM 1: Non-existent properties
+    // ==========================================
+
+    [Test]
+    [Category("Problem2")]
+    public void Problem2_NullValue_ShouldUseZeroSection()
+    {
+        var testObject = new { Value = (int?)null };
+        const string template = "{Value:positive;negative;zero}";
+        const string expected = "zero";
+
+        var actual = template.FormatWith(testObject, environment);
+        actual.ShouldBe(expected, "Null values should use zero section without transformation");
+    }
+
+    [Test]
+    [Category("Problem1")]
+    public void Problem1_MissingProperty_ShouldFailGracefully()
+    {
+        // Test tries to use {MajorMinorPatch} on SemanticVersion but that property doesn't exist
+        var semanticVersion = new SemanticVersion
+        {
+            Major = 1,
+            Minor = 2,
+            Patch = 3
+        };
+
+        const string template = "{MajorMinorPatch}"; // This property doesn't exist on SemanticVersion
+
+        // Currently this will throw or behave unexpectedly
+        // Should either throw meaningful error or handle gracefully
+        Assert.Throws<ArgumentException>(() => template.FormatWith(semanticVersion, environment));
```

Comment:
I think it's a good practice to test the returned `ArgumentException` to ensure that the right guard clause is throwing it. This just ensures that an `ArgumentException` is being thrown anywhere by the code path, not necessarily by our own code.